app/models/profile.rb

```diff
@@ -37,4 +32,26 @@ def self.static_fields
   def clear!
     update(data: {})
   end
+
+  # Lazily add accessors for profile fields on first use
+  def method_missing(method_name, *args, **kwargs, &block)
+    match = method_name.match(ATTRIBUTE_NAME_REGEX)
+    field = ProfileField.find_by(attribute_name: match[:attribute_name])
+    super unless field
+
+    self.class.instance_eval do
+      store_attribute :data, field.attribute_name.to_sym, field.type
```

Comment:
this lazy load within method_missing solves the refresh_attributes! issue of repeated delegation of the same fields (the stack trace issue).

Reply:
Assuming this is merged we should remove the `RUBY_THREAD_VM_STACK_SIZE` override in the travis config.

Reply:
I removed this from `.travis.yml` as part of this PR now.